pkg/uhttp/gocache.go

```diff
@@ -0,0 +1,170 @@
+package uhttp
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"time"
+
+	bigcache "github.com/allegro/bigcache/v3"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+)
+
+type GoCache struct {
+	ttl         time.Duration
+	rootLibrary *bigcache.BigCache
+}
+
+func NewGoCache(ctx context.Context, ttl int32, cacheMaxSize int, isLogLevel bool) (GoCache, error) {
+	l := ctxzap.Extract(ctx)
+	config := bigcache.Config{
+		// number of shards (must be a power of 2)
+		Shards: 1024,
```

Comment:
I don't think we need that many shards. 4 or 8 is probably enough. Higher shard counts are only needed for high concurrency, and we won't have that many outstanding http requests.

Reply:
Fixed